tracing-appender/src/worker.rs

```diff
@@ -74,10 +64,10 @@ impl<T: Write + Send + Sync + 'static> Worker<T> {
     /// Creates a worker thread that processes a channel until it's disconnected
     pub(crate) fn worker_thread(mut self) -> std::thread::JoinHandle<()> {
         thread::spawn(move || {
-            while !self.shutdown_signal.load(Ordering::Relaxed) {
+            loop {
                 match self.work() {
                     Ok(WorkerState::Continue) | Ok(WorkerState::Empty) => {}
-                    Ok(WorkerState::Disconnected) => break,
+                    Ok(WorkerState::Shutdown) | Ok(WorkerState::Disconnected) => break,
```

Comment:
Shutting down the worker should probably trigger a flush?

Reply:
oh yeah, that's right

Reply:
Was also relying on the call to flush outside of the loop to handle it.

Reply:
What do you think of just leaving it as is? I feel it's cleaner the way it is now, on breaking of the loop a flush will be triggered.

Reply:
Oh, yeah, I forgot we flushed after breaking the loop; this is fine.